drizzlepac/hlautils/astrometric_utils.py

```diff
@@ -403,30 +354,44 @@ def extract_sources(img, **pars):
     bkg_estimator = MedianBackground()
     bkg = None
 
-    exclude_percentiles = [10,25,50,75]
+    exclude_percentiles = [10, 25, 50, 75]
     for percentile in exclude_percentiles:
         try:
             bkg = Background2D(imgarr, (50, 50), filter_size=(3, 3),
-                           bkg_estimator=bkg_estimator,
-                           exclude_percentile=percentile)
+                               bkg_estimator=bkg_estimator,
+                               exclude_percentile=percentile)
+        except Exception:
+            bkg = None
+            continue
+
+        if bkg is not None:
             # If it succeeds, stop and use that value
             bkg_rms = (5. * bkg.background_rms)
             bkg_rms_mean = bkg.background.mean() + 5. * bkg_rms.std()
             default_threshold = bkg.background + bkg_rms
-            if threshold is None or threshold < 0.0:
-                if threshold is not None and threshold < 0.0:
-                    threshold = -1*threshold*default_threshold
-                    log.info("{} based on {}".format(threshold.max(), default_threshold.max()))
-                    bkg_rms_mean = threshold.max()
-                else:
-                    threshold = default_threshold
```

Comment:
This was a bit obscure to me. A suggestioon is below but it needs to be double checked that it reproduces the desired functionality.

Reply:
Your changes seem like they cover all the cases with much simpler logic.